swarm/src/lib.rs

```diff
@@ -695,7 +695,8 @@ where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
                                 if this.network.is_disconnected(&peer_id) => true,
                             DialPeerCondition::NotDialing
                                 if !this.network.is_dialing(&peer_id) => true,
-                            _ => false
+                            DialPeerCondition::Always => true,
+                            _ => false,
```

Comment:
It may be a good idea to make the match exhaustive, which should be possible since `DialPeerCondition` is defined in the same crate and would have prevented this bug in the first place, e.g. something like
```rust
let condition_matched = match condition {
    DialPeerCondition::Disconnected => this.network.is_disconnected(&peer_id),
    DialPeerCondition::NotDialing => !this.network.is_dialing(&peer_id),
    DialPeerCondition::Always => true,
};
```

Reply:
Agreed, just a second.